### PR TITLE
Adopt admiral API changes

### PR DESCRIPTION
**Pull request by tpantelis**
The addition of the `numRequeues` param to callback functions  and changes to broker syncer creation.

### CODE REVIEW

**Review by aswinsuryan — APPROVED** [review]

**Review by skitt — CHANGES_REQUESTED** [review]
Please run `go mod tidy`.

**Comment by tpantelis** [comment]
> Please run `go mod tidy`.

Done.

**Comment by tpantelis** [comment]
https://github.com/submariner-io/admiral/pull/172 also adds API changes so we can wait on this until after that is merged

**Review by skitt — APPROVED** [review]

**Pull request merged**
